test/common/stats/allocator_impl_test.cc

```diff
@@ -524,6 +524,65 @@ TEST_F(AllocatorImplTest, ForEachSinkedGauge) {
   EXPECT_EQ(num_iterations, 0);
 }
 
+TEST_F(AllocatorImplTest, ForEachSinkedGaugeHidden) {
+  GaugeSharedPtr sinked_gauge;
+  GaugeSharedPtr unsinked_gauge;
+
+  auto unhidden_stat_name = makeStat(absl::StrCat("unhidden.gauge"));
+  auto hidden_stat_name = makeStat(absl::StrCat("hidden.gauge"));
+
+  size_t num_gauges = 0;
+  size_t num_iterations = 0;
+
+  sinked_gauge =
+      alloc_.makeGauge(unhidden_stat_name, StatName(), {}, Gauge::ImportMode::Accumulate);
+
+  unsinked_gauge =
+      alloc_.makeGauge(hidden_stat_name, StatName(), {}, Gauge::ImportMode::HiddenAccumulate);
+
+  alloc_.forEachSinkedGauge([&num_gauges](std::size_t size) { num_gauges = size; },
+                            [&num_iterations, unhidden_stat_name](Gauge& gauge) {
+                              EXPECT_EQ(unhidden_stat_name, gauge.statName());
+                              num_iterations++;
+                            });
+  EXPECT_EQ(num_gauges, 2);
+  EXPECT_EQ(num_iterations, 1);
+}
+
+TEST_F(AllocatorImplTest, ForEachSinkedGaugeHiddenPredicate) {
+  std::unique_ptr<TestUtil::TestSinkPredicates> moved_sink_predicates =
+      std::make_unique<TestUtil::TestSinkPredicates>();
+  TestUtil::TestSinkPredicates* sink_predicates = moved_sink_predicates.get();
+  GaugeSharedPtr sinked_gauge;
+  GaugeSharedPtr unsinked_gauge;
+
+  alloc_.setSinkPredicates(std::move(moved_sink_predicates));
+
+  auto unhidden_stat_name = makeStat(absl::StrCat("unhidden.gauge"));
+  auto hidden_stat_name = makeStat(absl::StrCat("hidden.gauge"));
+
+  sink_predicates->add(unhidden_stat_name);
+  sink_predicates->add(hidden_stat_name);
+
+  size_t num_gauges = 0;
+  size_t num_iterations = 0;
+
+  sinked_gauge =
+      alloc_.makeGauge(unhidden_stat_name, StatName(), {}, Gauge::ImportMode::Accumulate);
+
+  unsinked_gauge =
```

Comment:
Can you clean up the variable naming here? I think the variable names reflect the previous semantics where the 'hidden' bit would override the predicate.